website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,764 @@
+        if (confirmed) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {}).then(function() {
+                bootbox.dialog({
+                    closeButton: false,
+                    message: afterSubmitForApprovalMessage,
+                    title: 'Pre-Registration Prize Submission',
+                    buttons: {
+                        registrations: {
+                            label: 'Return to registrations page',
+                            className: 'btn-primary pull-right',
+                            callback: function() {
+                                window.location.href = self.draft().urls.registrations;
+                            }
+                        }
+                    }
+                });
+            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
+        }
+    });
+};
+RegistrationEditor.prototype.submit = function() {
+    var self = this;
+    var currentNode = window.contextVars.node;
+    var currentUser = $osf.currentUser();
+    var messages = self.draft().messages;
+    bootbox.confirm(messages.beforeSubmitForApproval, function(result) {
+        if (result) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {
+                node: currentNode,
+                auth: currentUser
+            }).then(function() {
+                bootbox.dialog({
+                    message: messages.afterSubmitForApproval,
+                    title: 'Pre-Registration Prize Submission',
+                    buttons: {
+                        registrations: {
+                            label: 'Return to registrations page',
+                            className: 'btn-primary pull-right',
+                            callback: function() {
+                                window.location.href = self.draft().urls.registrations;
+                            }
+                        }
+                    }
+                });
+            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
+        }
+    });
+};
+/**
+ * Create a new draft
+ **/
+RegistrationEditor.prototype.create = function(schemaData) {
+    var self = this;
+
+    var metaSchema = self.draft().metaSchema;
+
+    return $osf.postJSON(self.urls.create, {
+        schema_name: metaSchema.name,
+        schema_version: metaSchema.version,
+        schema_data: schemaData
+    }).then(self.updateData.bind(self));
+};
+RegistrationEditor.prototype.putSaveData = function(payload) {
+    var self = this;
+    $osf.putJSON(self.urls.update.replace('{draft_pk}', self.draft().pk), payload).then(self.updateData.bind(self));
+};
+/**
+ * Save the current Draft
+ **/
+RegistrationEditor.prototype.save = function() {
+    var self = this;
+    var metaSchema = self.draft().metaSchema;
+    var schema = metaSchema.schema;
+    var data = {};
+    $.each(schema.pages, function(i, page) {
```

Comment:
It may make sense to separate the act of restructuring data from the act of sending the request. (separate methods)